libraries/WDT/examples/Example2_WDT_Config/Example2_WDT_Config.ino

```diff
@@ -0,0 +1,103 @@
+/*
+  Author: Adam Garbo
+  Created: August 1st, 2020
+  License: MIT. See SparkFun Arduino Apollo3 Project for more information
+
+  This example demonstrates how to modify the configuration of the Artemis
+  Watchdog Timer (WDT).
+
+  The watchdog timer is controlled by a clock divider, interrupt ticks and 
+  reset ticks. To achieve desired watchdog timing, a simple calculation can be made:
+
+  period = # ticks / clock divider frequency 
+
+  Examples: 
+  128 interrupt ticks / 128 Hz clock = 1 second
+  64 interrupt ticks / 16 Hz clock = 4 seconds
+  32 interrupt ticks / 1 Hz clock = 32 seconds
+  16 interrupt ticks / 1/16 Hz clock = 256 seconds
+  
+  The following code will configure the watchdog for both interrupt and reset
+  generation, and immediately start the watchdog timer.
+  The watchdog ISR provided will 'pet' the watchdog four times. On the fifth
+  interrupt, the watchdog will not be pet, so the 'reset' action can occur.
+  On the sixth timeout event, the WDT will issue a system reset, and the
+  program will start over from the beginning.
+
+  This example is based on the Ambiq SDK 2.4.2 watchdog.c example.
+
+  Tested with SparkFun Artemis Redboard.
+*/
+
+#include <WDT.h>
+
+APM3_WDT wdt;
+
+volatile bool watchdogFlag = false; // Watchdog Timer ISR flag
+volatile int watchdogInterrupt = 0; // Watchdog interrupt counter
+
+unsigned long currentMillis = 0,
+              previousMillis = 0;
+
+void setup()
+{
+  Serial.begin(115200);
+
+  Serial.println("Artemis Watchdog Timer Example");
+
+  // Configure the watchdog
+  /*
+    Available watchdog timer clock dividers:
+    0 = Low Power Mode. This setting disables the watch dog timer
```

Comment:
Oh damn, I was going backwards and only just found this.  I still think it's better to use macros, even if they are newly defined simpler ones rather than the longer HAL ones, to feed in as clock dividers.  Ie. #define WDT_128HZ, etc.

Reply:
It was interesting to go down the rabbit hole to find out more about the definitions. The HAL is ultimately just a definition of an enumeration in apollo3.h. I agree that simpler definitions may be more intuitive (e.g. 128HZ) and will look at adding some. The advantage is users are free to choose to use any of these three options (i.e. simple definition, HAL definition, or apollo3.h enumeration).

watchdog.cpp
```
#define AM_HAL_WDT_LFRC_CLK_128HZ   (_VAL2FLD(WDT_CFG_CLKSEL, WDT_CFG_CLKSEL_128HZ))
```

apollo3.h
```
typedef enum {                                  /*!< WDT_CFG_CLKSEL                                                            */
  WDT_CFG_CLKSEL_OFF                   = 0,     /*!< OFF : Low Power Mode. This setting disables the watch dog timer.          */
  WDT_CFG_CLKSEL_128HZ                 = 1,     /*!< 128HZ : 128 Hz LFRC clock.                                                */
  WDT_CFG_CLKSEL_16HZ                  = 2,     /*!< 16HZ : 16 Hz LFRC clock.                                                  */
  WDT_CFG_CLKSEL_1HZ                   = 3,     /*!< 1HZ : 1 Hz LFRC clock.                                                    */
  WDT_CFG_CLKSEL_1_16HZ                = 4,     /*!< 1_16HZ : 1/16th Hz LFRC clock.                                            */
} WDT_CFG_CLKSEL_Enum;
```